src/Http/TracingHttpClient.php

```diff
@@ -22,25 +22,24 @@
 use Symfony\Component\HttpClient\HttpClient;
 use Symfony\Component\HttpClient\HttpClientTrait;
 use Symfony\Component\HttpClient\Response\ResponseStream;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
 use Symfony\Contracts\HttpClient\HttpClientInterface;
 use Symfony\Contracts\HttpClient\ResponseInterface;
 use Symfony\Contracts\HttpClient\ResponseStreamInterface;
 
-final class TracingHttpClient implements HttpClientInterface
+class TracingHttpClient implements HttpClientInterface
```

Comment:
this is because decorating the HttpClient allows more customization on the service side (accessing things from service-provider isn't helpful). The constructor is now `final` to complies with phpstan's safe static usage rule.